src/Plugins/ComplexCore/src/ComplexCore/Filters/ImportHDF5Dataset.cpp

```diff
@@ -47,13 +47,17 @@ Result<> fillDataArray(DataStructure& dataStructure, const DataPath& dataArrayPa
 {
   auto& dataArray = dataStructure.getDataRefAs<DataArray<T>>(dataArrayPath);
   auto& absDataStore = dataArray.getDataStoreRef();
-  auto& dataStore = dynamic_cast<DataStore<T>&>(absDataStore);
-  if(!datasetReader.readIntoSpan<T>(dataStore.createSpan()))
+
+  std::vector<T> data(absDataStore.getSize());
```

Comment:
With ImportHDF5Dataset currently, I think it should only handle reading into an in memory data array. Since if the data is large enough to warrant out of core, then creating a std::vector here defeats the purpose. And we can avoid having to temporarily have to store double the data storage and do an extra copy for in memory data.